mrjob/bin.py

```diff
@@ -844,14 +844,14 @@ def _interpolate_spark_script_path(self, path):
 
     def _spark_cmdenv(self, step_num):
         """Returns a dictionary mapping environment variable to value,
-        including mapping PYSPARK_PYTHON to self._python_bin()
+        including mapping PYSPARK_PYTHON to self._task_python_bin()
         """
         step = self._get_step(step_num)
 
         cmdenv = {}
 
-        if step['type'] in ('spark', 'spark_script'):  # not spark_jar
-            cmdenv = dict(PYSPARK_PYTHON=cmd_line(self._python_bin()))
+        if step['type'] in ('spark', 'spark_script', 'spark_jar')
+            cmdenv = dict(PYSPARK_PYTHON=cmd_line(self._task_python_bin()))
```

Comment:
We dont need python in spark jar step...

Reply:
That was my guess. I just wanted have the code mirror what we end up doing in the TC code, which ends up adding these args.